apps/api/src/domain/models/context/edge-connection.entity.ts

```diff
@@ -234,14 +244,43 @@ export class EdgeConnection extends Aggregate {
         });
     }
 
+    @UpdateMethod()
+    addAudioForNote(
+        audioItemId: AggregateId,
+        languageCode: LanguageCode
+    ): ResultOrError<EdgeConnection> {
+        if (!this.note.has(languageCode)) {
+            return new CannnotAddAudioForNoteInGivenLanguageError(
+                this.id,
+                audioItemId,
+                languageCode
+            );
+        }
+
+        const updatedAudio = this.audioForNote.addAudio(audioItemId, languageCode);
+
+        // TODO remove cast
```

Comment:
Do you have an invalid case for when the audio to add is invalid in `edge-connection.add-audio-for-note.spec.ts`?